File: server/src/main/java/com/example/server/repository/entity/Item.java
package com.example.server.repository.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@AllArgsConstructor
@Data
@Entity
@NoArgsConstructor
@Table(name = "items")
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;

    @Column(name = "is_available")
    private Boolean available;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;

    @ManyToOne
    @JoinColumn(name = "request_id")
    private ItemRequest request;

    public boolean hasSameOwner(Item item) {
        return getOwner().hasSameId(item.getOwner());
    }

    public boolean hasSameOwner(long ownerId) {
        return getOwner().hasSameId(ownerId);
    }

    public boolean isAvailable() {
        return this.available;
    }
}

File: application/src/main/kotlin/com/example/server/entity/Item.kt
package com.example.server.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table

@Entity
@SequenceGenerator(name = "item_id_seq", allocationSize = 1)
@Table(name = "item")
class Item(

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "item_id_seq")
    var id: Long?,

    @Column(name = "name")
    var name: String,

    @Column(name = "description")
    var description: String,

    @Column(name = "is_available")
    var available: Boolean,

    @JoinColumn(name = "owner_id")
    @ManyToOne
    var owner: User,

    @JoinColumn(name = "request_id")
    @ManyToOne
    var request: ItemRequest?
)
